src/gui/parameters_screen.py

```diff
@@ -326,28 +418,257 @@ def validate_point_side_size(self):
+            self.oil_sources_listbox_insert.config(state=DISABLED)
+            self.correctly_set_parameters_oil_sources[3] = 0
+
+        def validate_spill_end_oil_source(self):
+            value = self.spill_end_oil_source_input.get()
+            if not value:
+                return False
+            if re.match('\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}', value):
+                try:
+                    pd.Timestamp(value)
+                    self.spill_end_oil_source = value
+                    self.spill_end_oil_source_validation_label.config(text="Valid value", fg="black")
+                    self.correctly_set_parameters_oil_sources[4] = 1
+                    self.check_all_parameters_validity_oil_sources()
+                    return True
+                except ValueError:
+                    pass
+            self.spill_end_oil_source_validation_label.config(text="Invalid value", fg="red")
+            self.oil_sources_listbox_insert.config(state=DISABLED)
+            self.correctly_set_parameters_oil_sources[4] = 0
+
+        def check_all_parameters_validity_oil_sources(self):
+            if sum(self.correctly_set_parameters_oil_sources) == 5:
+                self.oil_sources_listbox_insert.config(state=NORMAL)
+
         def validate_all_parameters(self):
-            self.validate_coordinates_top()
-            self.validate_coordinates_down()
-            self.validate_coordinates_left()
-            self.validate_coordinates_right()
+            self.validate_coordinates_top(False)
+            self.validate_coordinates_down(False)
+            self.validate_coordinates_left(False)
+            self.validate_coordinates_right(False)
             self.validate_time_range_start()
             self.validate_time_range_end()
             self.validate_data_time_step()
             self.validate_cells_side_count_latitude()
             self.validate_cells_side_count_longitude()
             self.validate_point_side_size()
-
-        def check_all_parameters_validity_and_refresh_image(self):
-            if sum(self.correctly_set_parameters[:4]) == 4:
-                self.load_and_crop_image()
-                if sum(self.correctly_set_parameters) == 10:
+            self.validate_iter_as_sec()
+            self.validate_min_oil_thickness()
+            self.validate_oil_viscosity()
+            self.validate_oil_density()
+
+        def validate_all_parameters_oil_sources_listbox(self):
+            self.validate_latitude_oil_source()
+            self.validate_longitude_oil_source()
+            self.validate_mass_per_minute_oil_source()
+            self.validate_spill_start_oil_source()
+            self.validate_spill_end_oil_source()
+
+        def check_all_parameters_validity_and_refresh_image(self, coordinate_change=False):
+            if coordinate_change and sum(self.correctly_set_parameters[:4]) == 4:
+                if sum(self.correctly_set_parameters) == 14:
```

Comment:
maybe "if all(self.correctly_set_parameters)"?  After changing number of parameters it will still work